src/index.js

```diff
@@ -335,10 +337,63 @@ class JanusAdapter {
     });
   }
 
+  toggleFreeze() {
+    if(this.frozen) {
+      this.unfreeze();
+    } else {
+      this.freeze();
+    }
+  }
+
+  freeze() {
+    this.frozen = true;
+  }
+
+  unfreeze() {
+    this.frozen = false;
+    this.flushPendingUpdates();
+  }
+
+  flushPendingUpdates() {
+    for (let [networkId, message] of this.frozenUpdates) {
+        if(!message.data.owner || this.occupants[message.data.owner]) { // ignore users who have disconnected since freezing, since their entities will have already been removed by NAF
```

Comment:
Huh. In what cases does a message not have an owner? And do we really want to discard those messages?

Reply:
delete messages do not have owners, and we arent discarding them we are keeping them if they have no owner. 

The point of this check is just to discard messages from users who have disconnected while we are frozen since NAF has already destroyed those entities, but we ahve stored state for them. If we allowed those to go through we would end up re-instantiating them in some cases (if the messages happened to contain the full state of the entity, which would be considered a "full sync") or at the very least just confusing NAF since it got an update for an entity that doesnt exist.